### DISCUSSION TITLE
Is there a block for turning off the hub and a block to keep scanning for a remote until connected?

### DISCUSSION THREAD

**Discussion by EmperorKabir** [pybricks/support#1404] [Q&A] [answered]
Hi, is there a block code for turning off a city hub? My intention is to map this function to the Lego remote as a killswitch so that I don't have to go and physically turn the hub(s) off. I'm aware that I would need to go and manually turn them back on again but I want to at least be able to broadcast from one hub to another to turn the receiving hub off then turn the main hub off. The Lego remote will then automatically power down after a few seconds automatically anyway.

Secondly, is it possible to force a hub to continually rescan for a remote until it finds one to connect to? This would be an if statement repeating until a remote is connected after which the main program runs. I often need to keep pressing the hub button and remote button just to get the timing right for it it connect up. At least with this functionality, I only need to do it on the remote.

(Third side question but I think I know the answer.... Presumably there is no functionality to account for either actual time of day or time elapsed? I'm guessing the city hub can't store or actually maintain this data between power cycles for actual time of day and that there are workarounds for time elapsed by using wait functions. E.g. turn off city hub at 9pm or e.g. stop function after 6 seconds).

**Comment by laurensvalk** [comment, marked as answer]
> Hi, is there a block code for turning off a city hub?

Good point. See new issue in https://github.com/pybricks/support/issues/1405

> Secondly, is it possible to force a hub to continually rescan for a remote until it finds one to connect to?

In the Python API, we have a timeout parameter that you can set to infinite. It defaults to 10 seconds in the blocks now, but for reasons similar to yours it's been set to infinite in the upcoming block version.

> I often need to keep pressing the hub button and remote button just to get the timing right for it it connect up. 

This should also be improved in the upcoming version. See https://github.com/pybricks/support/issues/880.

>  Presumably there is no functionality to account for either actual time of day or

No, the hub has no system clock.

> time elapsed? 

Yes! There is the StopWatch.

So you could make portions of your code run at certain times, as long as your batteries last :smile: 

![image](https://github.com/pybricks/support/assets/12326241/7a1e0eae-4f01-429a-8aff-02779de3ca14)

**Reply by EmperorKabir** [reply]
Thank you!
